src/usage_metrics/scripts/save_zenodo_metrics.py

```diff
@@ -0,0 +1,109 @@
+"""This script pull github traffic metrics and saves them to a GC Bucket."""
+
+import logging
+import sys
+from datetime import date, datetime
+from typing import Annotated
+
+import pandas as pd
+import requests
+from google.cloud import storage
+from pydantic import BaseModel, StringConstraints
+
+Doi = Annotated[str, StringConstraints(pattern=r"10\.5281/zenodo\.\d+")]
+SandboxDoi = Annotated[str, StringConstraints(pattern=r"10\.5072/zenodo\.\d+")]
+
+logger = logging.getLogger()
+logging.basicConfig(level="INFO")
+
+
+class ZenodoStats(BaseModel):
+    """Pydantic model representing Zenodo usage stats.
+
+    See https://developers.zenodo.org/#representation.
+    """
+
+    downloads: int
+    unique_downloads: int
+    views: int
+    unique_views: int
+    version_downloads: int
+    version_unique_downloads: int
+    version_unique_views: int
+    version_views: int
+
+
+class CommunityMetadata(BaseModel):
+    """Pydantic model representing Zenodo deposition metadata from the communities endpoint.
+
+    See https://developers.zenodo.org/#representation.
+    """
+
+    created: datetime = None
+    modified: datetime = None
+    recid: str
+    conceptrecid: str
+    doi: Doi | SandboxDoi | None = None
+    conceptdoi: Doi | SandboxDoi | None = None
+    doi_url: str
+    title: str
+    updated: datetime = None
+    stats: ZenodoStats
+
+    @classmethod
+    def check_empty_string(cls, doi: str):  # noqa: N805
+        """Sometimes zenodo returns an empty string for the `doi`. Convert to None."""
+        if doi == "":
+            return
+
+
+def get_zenodo_logs() -> pd.DataFrame():
+    """Get all metrics for all versions of records in the Catalyst Cooperative Zenodo community."""
+    community_url = "https://zenodo.org/api/communities/14454015-63f1-4f05-80fd-1a9b07593c9e/records"
+    community_records = requests.get(community_url, timeout=100)
+    catalyst_records = community_records.json()["hits"]["hits"]
+    catalyst_records = [CommunityMetadata(**record) for record in catalyst_records]
+    stats_dfs = []
+
+    for record in catalyst_records:
+        logger.info(f"Getting usage metrics for {record.title}")
+        # For each record in the community
+        versions_url = f"https://zenodo.org/api/records/{record.recid}/versions"
+        record_versions = requests.get(versions_url, timeout=100)
+        version_records = record_versions.json()["hits"]["hits"]
+        version_records = [CommunityMetadata(**record) for record in version_records]
+        version_df = pd.DataFrame(
+            [
+                dict(
+                    version_records[item].stats.__dict__,
+                    doi=version_records[item].doi,
+                    title=version_records[item].title,
+                )
+                for item in range(len(version_records))
+            ]
+        )
```

Comment:
You could simplify this by using the objects in the iterator instead of indexing:

```python
version_df = pd.DataFrame(
            [
                dict(
                    version_record.stats.dict(), # pydantic models have a dict() method you can use instead of accessing __dict__ directly
                    doi=version_record.doi,
                    title=version_record.title,
                )
                for version_record in version_records
            ]
        )
```